templates/components/listings/publications/tr_card.html.twig

```diff
@@ -37,6 +37,15 @@
         </div>
     {% endif %}
     {% endblock translations %}
+    {% block crs %}
+        {% set cr = spec.specification.lastSpecVersionOfType('CR') %}
+        {% if spec.crd and cr %}
+            <div>
+                <dt>Latest Candidate Recommendation Snapshot</dt>
+                <dd><a href="{{ cr.uri }}">{{ w3c_date_format(cr.date)|raw }}</a></dd>
+            </div>
+        {% endif %}
```

Comment:
No need to search for the latest CRS of that specification if the version is not a CRD

```suggestion
        {% if spec.crd %}
            {% set cr = spec.specification.lastSpecVersionOfType('CR') %} %}
            {% if cr %}
                <div>
                    <dt>Latest Candidate Recommendation Snapshot</dt>
                    <dd><a href="{{ cr.uri }}">{{ w3c_date_format(cr.date)|raw }}</a></dd>
                </div>
            {% endif %}
        {% endif %}
```